### PR TITLE
Jetpack Pro Dashboard: implement validation for notification settings form

### PR DESCRIPTION
**Pull request by yashwin**
#### Proposed Changes

This PR implements validation for the notification settings form in the Jetpack Pro Dashboard.

#### Testing Instructions

**Prerequisites**

Since these changes are made specifically for agencies, you must set yourself(partner) as an agency - 2c49b-pb. Make sure to switch it back to the previous type.

> **_NOTE:_** Changing the monitor status will not update the status immediately; we will have to debug this - 1203448324265423-as-1203576555374768. API integration to save the notification settings will be done in a separate PR.

**Instructions**

1. Run `git checkout add/validation-for-notification-settings-form` and `yarn start-jetpack-cloud`
2. Open http://jetpack.cloud.localhost:3000/, and you'll be redirected to the /dashboard.
3. Click the clock icon(which also shows the current notification schedule) or Enable the monitor for any site by clicking the toggle on the monitor column if you don't have monitor enabled on any of your sites.
4. Please note the monitor will not be enabled immediately; you might have to wait for some time until it is enabled. 
5. Once clicked on the clock icon, the notification settings popup will be opened.
6. Set both the toggle off for `Mobile` & `Email` -> Click `Save` -> Verify that an error message is shown as below and the `Save` button is disabled. Setting the toggle on for `Mobile` or `Email` should remove the error message and enable the `Save` button.

Screenshot 2022-12-22 at 12 50 41 PM

Screenshot 2022-12-22 at 12 51 58 PM

7. Set toggle on for `Email`and don't enter any email id -> Click `Save` -> Verify that an error message is shown as below and the `Save` button is disabled. Adding an email id should remove the error message and enable the `Save` button.

Screenshot 2022-12-23 at 5 00 35 PM

Screenshot 2022-12-23 at 4 59 57 PM

#### Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Related to 1203448324265423-as-1203507036638804

### CODE REVIEW

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/notification-settings/index.tsx [hunk 1]
```diff
@@ -54,12 +61,25 @@ export default function NotificationSettings( { onClose, site, settings }: Props
...
+		if ( recipients.length ) {
+			setValidationError( '' );
+		}
+		setAddedEmailAddresses( recipients );
+	}
+
 	const addEmailsContent = enableEmailNotification && (
 		<div className="notification-settings__email-container">
 			<TokenField
 				placeholder="Enter Email IDs"
```
Comment (madebynoam) [inline client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/notification-settings/index.tsx]: Just a minor note here. Could we change it to "Enter email address"? Should this use the translate function?

**Comment by madebynoam** [comment]
Lgtm, thanks @yashwin! 🙌 I just left a small note on a string (in case you want to change it in this PR).

File: client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/notification-settings/index.tsx (hunk 1 above)
Comment (yashwin) [reply client/jetpack-cloud/sections/agency-dashboard/downtime-monitoring/notification-settings/index.tsx]: Yes, I missed this. Updated here - https://github.com/Automattic/wp-calypso/pull/71479/commits/afe6d08ed3b2a5c8b5995231df323806ba804ec9

**Comment by madebynoam** [comment]
Perfect, thanks, @yashwin! lgtm!

**Review by madebynoam — APPROVED** [review]

**Review by vitozev — COMMENTED** [review]
In this case, is it expected to see the error below? I chose "Email" as the notification type, and a valid email address is entered in the input.

image

Are we planning to add a custom validator for a valid email address? We don't want to accept values such as `xyz` as valid ones.

**Comment by yashwin** [comment]
Thanks for the review, @vitozev!

> In this case, is it expected to see the error below? I chose "Email" as the notification type, and a valid email address is entered in the input.

I missed pushing this fix in this branch. It's fixed now - https://github.com/Automattic/wp-calypso/pull/71479/commits/47a1a89b65c8feaa1c4bdda913c716f47a5119aa.

> Are we planning to add a custom validator for a valid email address? We don't want to accept values such as `xyz` as valid ones.

Good point. We will address this in the coming days - 1203448324265423-as-1203634022020882

**Review by vitozev — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7706658

Thank you @yashwin for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7706658).

**Pull request merged**